skafka/src/main/scala/com/evolutiongaming/skafka/consumer/RebalanceListener1.scala

```diff
@@ -0,0 +1,77 @@
+package com.evolutiongaming.skafka.consumer
+
+import cats.data.NonEmptySet
+import cats.~>
+import com.evolutiongaming.skafka.TopicPartition
+
+/**
+  * Will be converted to [[org.apache.kafka.clients.consumer.ConsumerRebalanceListener]] during consumer.subscribe
+  *
+  * Uses [[RebalanceCallback]] to describe the actions to be performed during rebalance.
+  *
+  * Please refer to [[RebalanceCallback]] for more details.
+  *
+  * Below is an example inspired by [[org.apache.kafka.clients.consumer.ConsumerRebalanceListener]] documentation.
+  *
+  * Compiling and working example is available [[https://github.com/evolution-gaming/skafka/blob/d2af038b012523533f2b73d432721d6d1e7cebbe/skafka/src/test/scala/com/evolutiongaming/skafka/consumer/RebalanceListener1Spec.scala here]]:
+  * TODO: replace with shorter github link after merge of https://github.com/evolution-gaming/skafka/pull/122
+  * {{{
+  *
+  * class SaveOffsetsOnRebalance[F[_]: Applicative] extends RebalanceListener1[F] {
```

Comment:
I just thought, shouldn't we add some **predefined** implementations for our `RebalanceListener1` ? :)
like the one from example? wdyt?

```
RebalanceListener1. saveOffsetsOnRebalance
```

Reply:
good idea, but I would let it live as an idea for some time before actually implementing it.
so I think we should not include any default implementation of `RebalanceListener1` in scope of this PR